job-server/src/spark.jobserver/JarManager.scala

```diff
@@ -5,9 +5,12 @@ import spark.jobserver.io.JobDAO
 import spark.jobserver.util.JarUtils
 import org.joda.time.DateTime
 
+import java.nio.file.{Files, Paths}
+
 // Messages to JarManager actor
 case class StoreJar(appName: String, jarBytes: Array[Byte])
 case object ListJars
+case class StoreInitialJars(initialJars: Map[String, String])
```

Comment:
Very minor nit: from JarManager's perspective it doesn't care if they are initial or not, could call the message `StoreJars`.